meta-resin-common/classes/resin-sanity.bbclass

```diff
@@ -0,0 +1,22 @@
+#
+# Sanity checks for resinOS builds
+#
+# Copyright (C) 2017 resin.io
+# Author: Andrei Gherzan <andrei@resin.io>
+#
+# Licensed under the Apache-2.0 license, see COPYING.Apache-2.0 for details
+
+def resinos_build_configuration(fatals):
+	if d.getVar('PACKAGE_CLASSES', True) != "package_ipk":
+		bb.error("ResinOS distro depends on opkg packages (ipk). Make sure PACKAGE_CLASSES is set on package_ipk.")
+		fatals.append("PACKAGE_CLASSES")
+
+python resinos_sanity_handler() {
+	f = []
+	resinos_build_configuration(f)
+	if (f):
+		bb.fatal("ResinOS sanity checks failed. See above.")
+}
+
+addhandler resinos_sanity_handler
+resinos_sanity_handler[eventmask] = "bb.event.BuildStarted"
```

Comment:
Could the ConfigParsed event be used here instead to better reflect that this handler does a configuration check?